vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/rest/HtsgetVcfController.java

```diff
@@ -53,145 +57,178 @@
 
     private Properties evaProperties;
 
-    @Autowired
     private VariantSourceService variantSourceService;
-    @Autowired
+
     private VariantWithSamplesAndAnnotationsService variantService;
 
-    public HtsgetVcfController() throws IOException {
+    public HtsgetVcfController(VariantSourceService variantSourceService,
+                               VariantWithSamplesAndAnnotationsService variantService) throws IOException {
+        this.variantSourceService = variantSourceService;
+        this.variantService = variantService;
         evaProperties = new Properties();
         evaProperties.load(VcfDumperWSServer.class.getResourceAsStream("/eva.properties"));
     }
 
     @RequestMapping(value = "/{id}", method = RequestMethod.GET, consumes = "application/*",
             produces = "application/vnd.ga4gh.htsget.v0.2rc+json; charset=UTF-8")
     public ResponseEntity getHtsgetUrls(
+            @ApiParam(value = "Study identifiers (separate with comma for multiple studies), e.g. PRJEB9799. " +
+                    "Each individual identifier of studies can be looked up in " +
+                    "https://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all in the field named 'id'.",
+                    required = true)
             @PathVariable("id") String id,
+            @ApiParam(value = "Format in which the data will be represented, e.g. VCF", defaultValue = "VCF")
             @RequestParam(name = "format", required = false) String format,
+            @ApiParam(value = "Reference sequence name, e.g. 1 or chr1 or CM000001.1")
             @RequestParam(name = "referenceName", required = false) String referenceName,
+            @ApiParam(value = "First letter of the genus, followed by the full species name, e.g. ecaballus_20. " +
+                    "Allowed values can be looked up in https://www.ebi.ac.uk/eva/webservices/rest/v1/meta/species/list/" +
+                    " concatenating the fields 'taxonomyCode' and 'assemblyCode' (separated by underscore).",
+                    required = true)
             @RequestParam(name = "species", required = false) String species,
+            @ApiParam(value = "Start position, e.g. 3000000")
             @RequestParam(name = "start", required = false) Long start,
+            @ApiParam(value = "End position, e.g. 3010000")
```

Comment:
```suggestion
            @ApiParam(value = "End position (0-based exclusive), e.g. 3010000")
```

Reply:
Suggestion to comply with htsget protocol spec...

Reply:
`start` and `end` are both inclusive

Reply:
ok, I didn't see the spec has the `end` as exclusive. I will change that